includes/notifications.php

```diff
@@ -262,3 +276,42 @@ function pmpro_footer_link() {
 	<?php }
 }
 add_action( 'wp_footer', 'pmpro_footer_link' );
+
+/**
+ * Check if a plugin is active with a specific version.
+ * @param array $checks Plugin data to run a check. Includes: [ 0 => plugin-slug, 1 => comparison operator, 2 => version_to_check]
+ */
+function pmpro_check_plugin_version( $checks = NULL ) {
+	// Assume pass is failing right now.
+	$pass = false;
+
+	// run check if plugin data is not emptpy.
+	if ( $checks ) {
+		$plugin_file = $checks[0] . '/' . $checks[0] . '.php';
+		$plugin_data = get_plugin_data(  WP_PLUGIN_DIR . '/' . $plugin_file, false, true );
+
+		$conditional_check =  $checks[2] . " " . $checks[1] . " " . $plugin_data['Version'];
+
+		if ( $conditional_check ) {
```

Comment:
This will never be false (empty) even if the checks array is empty and the $plugin_data['Version'] is empty. It's minimally always a concatenated string of 2 spaces.

Reply:
I like that you're interested in this. :) Thanks for pointing these out.

This code is very much incomplete and we're abusing PRs to pass the code back and forth between andrew and I in a public way. I'm working on it more today and will reach out to you for a once over if you are interested. We might push it next week.